### PR TITLE
[stable/jenkins] Allow templating of ServiceAccount annotations

### PR DESCRIPTION
**Pull request by wmcdona89**
#### What this PR does / why we need it:
Allow templating of `serviceAccount.annotations` and `serviceAccountAgent.annotations` by rendering them with the `tpl` function

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
A use case for templating ServiceAccount annotations is the use of a [ServiceAccount as an OAuth client in OpenShift](https://docs.openshift.com/container-platform/4.3/authentication/using-service-accounts-as-oauth-client.html#redirect-uris-for-service-accounts_using-service-accounts-as-oauth-client). 

In this example, the Route name is templated so the OAuth redirect can point to a Route in the release rather than being pinned to a hardcoded Route.
```yaml
serviceAccount:
  annotations:
    serviceaccounts.openshift.io/oauth-redirectreference.jenkins: |-
      {
        "kind": "OAuthRedirectReference",
        "apiVersion": "v1",
        "reference": {
          "kind": "Route",
          "name": "{{ template "jenkins.fullname" . }}"
        }
      }
```
### helm template test
Given the above yaml in stable/jenkins/sa-values.yaml
```
helm template stable/jenkins \
--show-only templates/service-account.yaml \
--values stable/jenkins/sa-values.yaml
```
```yaml
# Source: jenkins/templates/service-account.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: RELEASE-NAME-jenkins
  namespace: default
  annotations:
    serviceaccounts.openshift.io/oauth-redirectreference.jenkins: |-
      {
        "kind": "OAuthRedirectReference",
        "apiVersion": "v1",
        "reference": {
          "kind": "Route",
          "name": "RELEASE-NAME-jenkins"
        }
      }
  labels:
    "app.kubernetes.io/name": 'jenkins'
    "helm.sh/chart": "jenkins-1.17.2"
    "app.kubernetes.io/managed-by": "Helm"
    "app.kubernetes.io/instance": "RELEASE-NAME"
    "app.kubernetes.io/component": "jenkins-master"
```

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/jenkins/templates/service-account.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ metadata:
   namespace: {{ template "jenkins.namespace" . }}
 {{- if .Values.serviceAccount.annotations }}
   annotations:
-{{ toYaml .Values.serviceAccount.annotations | indent 4 }}
-{{ end }}
+{{ tpl (toYaml .Values.serviceAccount.annotations) . | indent 4 }}
+{{- end }}
```
Comment (wmcdona89) [inline stable/jenkins/templates/service-account.yaml:10]: chomp newline

File: stable/jenkins/templates/service-account-agent.yaml [hunk 2]
```diff
@@ -6,8 +6,8 @@ metadata:
   namespace: {{ template "jenkins.master.slaveKubernetesNamespace" . }}
 {{- if .Values.serviceAccountAgent.annotations }}
   annotations:
-{{ toYaml .Values.serviceAccountAgent.annotations | indent 4 }}
-{{ end }}
+{{ tpl (toYaml .Values.serviceAccountAgent.annotations) . | indent 4 }}
+{{- end }}
```
Comment (wmcdona89) [inline stable/jenkins/templates/service-account-agent.yaml:10]: chomp newline

**Comment by torstenwalter** [comment]
@wmcdona89 Are you interested in becoming a maintainer of this chart?

**Comment by wmcdona89** [comment]
@torstenwalter Yes, I'm interested in helping maintain this chart. BTW...I'm nearing the end of my list of PRs to open and thanks for helping me get all of these merged :-)

**Comment by torstenwalter** [comment]
@wmcdona89 great to hear. Please add yourself as maintainer in Chart.yaml and OWNERS file.

**Comment by wmcdona89** [comment]
@torstenwalter I added myself.

**Review by torstenwalter — COMMENTED** [review]
Can we add an example which tests this in other-values.yaml? E.g. just adding a label with the name of the Jenkins instance to make use of templating?

**Comment by wmcdona89** [comment]
Added a templated description.
```
  annotations:
    description: "Used by release {{ .Release.Name }} for role-based access control"
```

**Review by torstenwalter — APPROVED** [review]
/lgtm

**Pull request merged**